tests/filecheck/dialects/fsm/fsm_op.mlir

```diff
@@ -0,0 +1,146 @@
+// RUN XDSL_ROUNDTRIP
+
+// CHECK:  "fsm.machine"() ({
+// CHECK:  ^bb0(%arg0: i1):
+// CHECK:    "fsm.state"() ({
+// CHECK:    }, {
+// CHECK:    }) {sym_name = "IDLE"} : () -> ()
+// CHECK:  }) {function_type = (i1) -> (), initialState = "IDLE", sym_name = "foo"} : () -> ()
+
+"builtin.module"() ({
```

Comment:
You can avoid an indentation level by omitting `builtin.module` if you want; it gets added automatically.